### ISSUE TITLE
Require or prohibit `break` after `default` in `switch`. 

### ISSUE THREAD

**Issue by jaepage** [eslint/eslint#7697]
**Please describe what the rule should do:**

Example of discussion: http://softwareengineering.stackexchange.com/questions/201777/break-on-default-case-in-switch

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
    switch (foo) {
        case 'bar':
            // Do something.
            break;
        default:
            // Do something else.
            break; // Or do we omit this break? What's our team's preference? 
    }
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It's a style choice that should be maintained across a project.

**Comment by platinumazure** [comment]
Is this rule proposal about requiring/prohibiting a `break` in the `default` case specifically, or in the last case even if it isn't default?

Does `no-fallthrough` help at all? (I'm guessing not, if it doesn't enforce anything in the last case statement, but I want to make sure you check it out.)

**Comment by jaepage** [comment]
This is a rule proposal to allow enforcement of whether `default` cases in `switch` statements should always or never have a `break` statement. 

I hadn't considered whether this would also apply to final cases in switches that have no default. I suppose it could. 

I do not believe the existing `no-fallthrough` rule would be applicable to this scenario.
